src/main/java/sparksoniq/jsoniq/item/DoubleItem.java

```diff
@@ -157,4 +162,44 @@ public Item compareItem(Item other, OperationalExpressionBase.Operator operator,
         }
         return operator.apply(this, other);
     }
+
+    @Override
+    public Item add(Item other) {
+        if (other.isDecimal())
```

Comment:
I think it is the other way round between decimal and double: for decimal and double, double wins. This is called type promotion https://www.w3.org/TR/xquery-30/#dt-type-promotion

Reply:
I think you are right. Thanks for spotting this!

Reply:
I notice that by doing this change, we get that `(1 + 3.14 + 10e1) * 0.1` results in `10.414000000000001` (just like in Master) because with doubles there is no such options as in BigDecimal to _stripTrailingZeros()_. There are two options:

1. We compute operations MUL, DIV, MOD in DoubleItem casting to Decimal and then stripping trailing zeros and then back to double like this:
` ItemFactory.getInstance().createDoubleItem(this.castToDecimalValue().multiply(other.castToDecimalValue()).doubleValue());
` for multiply() function in DoubleItem. I expect this to add some time overhead!
2. We accept the trailing zeros (or nines) as it is in master now

What do you think?

Reply:
Thank you for the thorough check.

The W3C specification relies on the IEEE 754-2008 standard for double arithmetics. This is an advanced topic and, for now, simply forwarding the behavior of double operations in Java as is will be fine. If Java says 10.414000000000001, then we can just return this.

This behavior is not a regression, but an improvement over the master in terms of correctness, so that it is fine to leave some fine-tuning for later.